client/my-sites/promote-post-i2/components/payments-filter/index.tsx

```diff
@@ -0,0 +1,76 @@
+import { SegmentedControl, SelectDropdown } from '@automattic/components';
+import { useMediaQuery } from '@wordpress/compose';
+import { translate } from 'i18n-calypso';
+import React, { useRef } from 'react';
+import { DropdownOption } from '../search-bar';
+import './style.scss';
+
+export type PaymentsFilterType = 'unified' | 'currentSite';
+
+interface Props {
+	options: DropdownOption[];
+	paymentsFilter: PaymentsFilterType;
+	handleChangeFilter: ( flag: boolean ) => void;
+}
+
+export default function PaymentsFilter( props: Props ) {
+	const tabsRef = useRef< { [ key: string ]: HTMLSpanElement | null } >( {} );
+	const onTabClick = ( key: string ) => {
+		tabsRef.current[ key ]?.scrollIntoView( {
+			behavior: 'smooth',
+			block: 'nearest',
+			inline: 'center',
+		} );
+	};
+
+	const { handleChangeFilter, paymentsFilter, options } = props;
+
+	const onChangePaymentFilterDropdown = ( option: string ) => {
+		handleChangeFilter( option === 'currentSite' );
+	};
+
+	const isDesktop = useMediaQuery( '(min-width: 1300px)' );
```

Comment:
Just checking on this, there are some existing functions for this that might do what you need here.

E.g. `useDesktopBreakpoint` or `useBreakpoint( '>=782px' );`

Reply:
we already use this here
https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/promote-post-i2/components/search-bar/index.tsx#L177

i would rather use the existing one so it is consistent in the blaze dash